Xamarin.PropertyEditing.Mac/Controls/EditorContainer.cs

```diff
@@ -24,6 +28,127 @@ public EditorViewModel ViewModel
 
 				if (PropertyButton != null)
 					PropertyButton.ViewModel = value as PropertyViewModel;
+
+				UpdateVariations (value as PropertyViewModel);
+			}
+		}
+
+		private void UpdateVariations (PropertyViewModel propertyViewModel)
+		{
+			if (propertyViewModel != null && propertyViewModel.Property.CanWrite) {
+				if (propertyViewModel.HasVariations) {
+
+					if (propertyViewModel.IsVariant) {
+						WantsLayer = true;
+
+						if (propertyViewModel.GetIsLastVariant ()) {
+							var endLineShape = new CAShapeLayer ();
+							var endLinePath = new NSBezierPath ();
+
+							endLinePath.MoveTo (new CGPoint (10, 35));
+							endLinePath.Append (new CGPoint[] { new CGPoint (10, 65) });
+							endLinePath.MoveTo (new CGPoint (10, 35));
+							endLinePath.Append (new CGPoint[] { new CGPoint (15, 35) });
+							endLineShape.Path = endLinePath.ToCGPath ();
+							endLineShape.FillColor = null;
+							endLineShape.Opacity = 1.0f;
+							endLineShape.StrokeColor = HostResources.GetNamedColor (NamedResources.ForegroundColor).CGColor;
+							Layer.AddSublayer (endLineShape);
+
+							this.variationLayersList.Add (endLineShape);
+						} else {
+							var midLineShape = new CAShapeLayer ();
+							var midLinePath = new NSBezierPath ();
+
+							midLinePath.MoveTo (new CGPoint (10, 0));
+							midLinePath.Append (new CGPoint[] { new CGPoint (10, 55) });
```

Comment:
Do we need to duplicate all this when these are the only two different lines?

Reply:
Yes I should have refactored that.